interfaces/Portal/src/app/pages/home/components/programs-list/programs-list.component.ts

```diff
@@ -27,10 +27,15 @@ export class ProgramsListComponent implements OnInit {
 
   private async refresh() {
     this.loading = true;
-    const programs = await this.programsService.getAllPrograms();
+    const programIds = await this.programsService.getAllProgramIds();
```

Comment:
@RubenGeo 

Weird idea: The Portal can already 'know' these Program-IDs by looking at the "currently logged-in user"'s permissions...
(So a method needs to be added to the AuthService/UserState-service; To return only the keys/indexes from the `permissions`-object)

That would prevent 1 full API-call... (and the corresponding back-end-code).

So this component would then "show all the details of the programs the user has access to"...

Reply:
Nice idea. I am just wondering if you would need to login in and out to see newly assigned programs. Or is that needed anyway to get access to them?

Reply:
We could use some "refresh the current user"-logic somewhere indeed.
But there is already the "current-user"-endpoint... so that is probably useful for it.

Reply:
Maybe on refresh of the programs page. But maybe that is scope creep for this PR @elwinschmitz

Reply:
@RubenGeo Its only these changes:
https://github.com/global-121/121-platform/compare/feat.performance-program-page...feat.add-current-user-check